pkg/ui/ui.go

```diff
@@ -0,0 +1,142 @@
+package ui
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io"
+	"math"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/common/route"
+	"github.com/prometheus/common/version"
+)
+
+type BaseUI struct {
+	logger    log.Logger
+	menuTmpl  string
+	tmplFuncs template.FuncMap
+}
+
+func NewBaseUI(logger log.Logger, menuTmpl string, funcMap template.FuncMap) *BaseUI {
+
+	funcMap["since"] = func(t time.Time) time.Duration {
+		return time.Since(t) / time.Millisecond * time.Millisecond
+	}
+	funcMap["pathPrefix"] = func() string { return "" }
```

Comment:
Are all of these really necessary for both rule and query? I think I am fine anyway, but maybe if some are used only in query or only in rule.. let's move to specific places.